great_expectations/profiler/domain_builder/domain.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import IDDict
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.execution_engine.execution_engine import MetricDomainTypes
+from great_expectations.types import SerializableDictDot
+
+
+class DomainTypes(Enum):
+    pass
```

Comment:
I still have some open questions about this (how does this relate to `MetricDomainTypes` and should `SemanticDomainTypes` continue to live in `SimpleSemanticTypeColumnDomainBuilder`) but I think this is a great step forward and we should continue the discussion using this code as a baseline.

Reply:
@anthonyburdi Yes, and I also will add a docstring saying that `DomainTypes` was created as the base class for the various kind of domain types, which, among other purposes serves to enable serialization into a JSON string.